ipatests/test_integration/test_automember.py

```diff
@@ -0,0 +1,366 @@
+
+    def test_modify_user_entry_unmembership(self):
+        # https://pagure.io/389-ds-base/issue/50077
+        """
+        Test that an update on a user entry triggers a re-evaluation of the
+        user's automember groups when auto-unmembership is active.
+        """
+        try:
+            tasks.kinit_admin(self.master)
+            master = self.master
+            search_cmd = [
+                'ldapsearch', '-x',
+                '-D', 'cn=directory manager',
+                '-w', master.config.dirman_password,
+                '-b', 'cn=Auto Membership Plugin,cn=plugins,cn=config',
+                '-s', 'base']
+            result = self.master.run_command(
+                search_cmd,
+                raiseonerr=False
+            )
+            assert result.returncode == 0
+            assert "automemberprocessmodifyops: on" in result.stdout_text
+
+            # testcase setup
+            group1 = 'DallasUsers'
+            group2 = 'HoustonUsers'
+            user1 = 'user1'
+            first = 'first'
+            last = 'user'
+            self.automember_group_add(group1)
+            self.automember_group_add(group2)
+            self.user_automember_setup(group1, user1, first, last)
+
+            result = self.master.run_command(
+                ['ipa', 'group-show', group1]
+            )
+            assert user1 in result.stdout_text
+
+            # Modifying the user group so that it becomes part of new group
+            self.user_automember_mod(group2, user1)
+
+            result = self.master.run_command(
+                ['ipa', 'group-show', group2]
+            )
+            assert user1 in result.stdout_text
+            result = self.master.run_command(
+                ['ipa', 'group-show', group1]
+            )
+            assert user1 not in result.stdout_text
+
+            # Deleting the user entry and checking its not part of group now
+            self.user_automember_cleanup(user1)
+
+            result = self.master.run_command(
+                ['ipa', 'group-show', group2]
+            )
+            assert user1 not in result.stdout_text
+
+        finally:
+            # testcase cleanup
+            self.automember_group_cleanup(group1)
+            self.automember_group_cleanup(group2)
+
+    def test_modify_host_entry_unmembership(self):
+        # https://pagure.io/389-ds-base/issue/50077
+        """
+        Test that an update on a host entry triggers a re-evaluation of the
+        host's automember groups when auto-unmembership is active.
+        """
+        try:
+            # testcase setup
+            hostgroup1 = 'Brno'
+            hostgroup2 = 'Boston'
+            host1 = 'host1.example.com'
+
+            self.automember_hostgroup_add(hostgroup1)
+            self.automember_hostgroup_add(hostgroup2)
+            self.host_automember_setup(hostgroup1, host1)
+
+            result = self.master.run_command(
```

Comment:
This snippet is repeated 9 times throughout the test suite. It is better to avoid repeating by adding a method like
def is_host_member_of_hostgroup(self, host, hostgroup):
  .....

Then this function can be invoked like
assert is_host_member_of_hostgroup(host, hostgroup)

or assert not is_host_member_of_hostgroup(host, hostgroup)